### PR TITLE
feat: add support for cluster user secrets and settings

### PR DESCRIPTION
**Pull request by aumer-amr**
This allows for adding extra cluster-secrets to your cluster, without any chance of overriding them accidently.
Since ansible will now always check if the file `cluster-secrets-extra.yaml` already exists or no.

### CODE REVIEW

**Comment by aumer-amr** [comment]
Note im deffo not an ansible person, so there are more likely better ways.

**Comment by onedr0p** [comment]
I have a few ideas / questions

1. How do you feel about `cluster-secrets-user` instead of `cluster-secrets-extra` ? and we could also do a `cluster-settings-user` to supplement this change.

2. I'm also curious if that we need to update the bootstrap task in taskfiles to include these new files?

File: .taskfiles/ClusterTasks.yaml [hunk 1]
```diff
@@ -14,6 +14,7 @@ tasks:
       - cat {{.SOPS_AGE_KEY_FILE}} | kubectl -n flux-system create secret generic sops-age --from-file=age.agekey=/dev/stdin
       - sops --decrypt {{.KUBERNETES_DIR}}/flux/vars/cluster-secrets.sops.yaml | kubectl apply -f -
       - kubectl apply -f {{.KUBERNETES_DIR}}/flux/vars/cluster-settings.yaml
+      - kubectl apply -f {{.KUBERNETES_DIR}}/flux/vars/cluster-settings-user.yaml
```
Comment (onedr0p) [inline .taskfiles/ClusterTasks.yaml:17]:
```suggestion
      - sops --decrypt {{.KUBERNETES_DIR}}/flux/vars/cluster-secrets.sops.yaml | kubectl apply -f -
      - sops --decrypt {{.KUBERNETES_DIR}}/flux/vars/cluster-secrets-user.sops.yaml | kubectl apply -f -
      - kubectl apply -f {{.KUBERNETES_DIR}}/flux/vars/cluster-settings.yaml
      - kubectl apply -f {{.KUBERNETES_DIR}}/flux/vars/cluster-settings-user.yaml
```

File: bootstrap/tasks/kubernetes/main.yaml [hunk 2]
```diff
@@ -8,7 +8,33 @@
...
 - name: Template Kubernetes unencrypted files
-  when: item.state == 'file' and 'sops' not in item.path and '.DS_Store' not in item.path
+  when: item.state == 'file' and 'sops' not in item.path and '.DS_Store' not in item.path and item.path not is match('cluster-(settings|secrets)-user\.yaml\.j2')
+  ansible.builtin.template:
+    src: "{{ item.src }}"
+    dest: "{{ repository_path }}/kubernetes/{{ item.path | regex_replace('.j2$', '') }}"
+    mode: "0644"
+  with_community.general.filetree: ["../templates/kubernetes/"]
+
+- name: Check if 'cluster-settings-user.yaml' already exists
```
Comment (onedr0p) [inline bootstrap/tasks/kubernetes/main.yaml:18]:
```suggestion
- name: Check if the cluster user settings file already exists
```

**Review by onedr0p — CHANGES_REQUESTED** [review]
Small nits here but overall looking good.

File: .vscode/settings.json [hunk 3]
```diff
@@ -16,5 +16,5 @@
     "editor.guides.highlightActiveBracketPair": true,
     "editor.hover.delay": 1500,
     "files.trimTrailingWhitespace": true,
-    "ansible.python.interpreterPath": "/usr/bin/python3",
+    "ansible.python.interpreterPath": "/home/linuxbrew/.linuxbrew/bin/python3",
```
[the remark this answers is not in the document]
Comment (onedr0p) [reply .vscode/settings.json:19]: We should just remove this line, VSCode keeps adding it in and I keep forgetting to remove it.

File: bootstrap/templates/kubernetes/flux/vars/cluster-secrets-user.sops.yaml.j2 [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cluster-secrets-extra
```
Comment (onedr0p) [inline bootstrap/templates/kubernetes/flux/vars/cluster-secrets-user.sops.yaml.j2:5]:
```suggestion
  name: cluster-secrets-user
```

File: bootstrap/templates/kubernetes/flux/apps.yaml.j2 [hunk 5]
```diff
@@ -21,6 +21,8 @@ spec:
         name: cluster-settings
       - kind: Secret
         name: cluster-secrets
+      - kind: Secret
+        name: cluster-secrets-extra
```
Comment (onedr0p) [inline bootstrap/templates/kubernetes/flux/apps.yaml.j2:25]:
```suggestion
      - kind: Secret
        name: cluster-secrets-user
      - kind: ConfigMap
        name: cluster-settings-user
```

File: bootstrap/templates/kubernetes/flux/apps.yaml.j2 [hunk 6]
```diff
@@ -38,6 +42,8 @@ spec:
                 name: cluster-settings
               - kind: Secret
                 name: cluster-secrets
+              - kind: Secret
+                name: cluster-secrets-extra
```
Comment (onedr0p) [inline bootstrap/templates/kubernetes/flux/apps.yaml.j2:46]:
```suggestion
              - kind: Secret
                name: cluster-secrets-user
              - kind: ConfigMap
                name: cluster-settings-user
```

File: bootstrap/templates/kubernetes/flux/apps.yaml.j2 [hunk 7]
```diff
@@ -21,6 +21,10 @@ spec:
         name: cluster-settings
       - kind: Secret
         name: cluster-secrets
+      - kind: Secret
+        name: cluster-secrets-user
+      - kind: ConfigMap
+        name: cluster-settings-user
```
Comment (onedr0p) [inline bootstrap/templates/kubernetes/flux/apps.yaml.j2:27]:
```suggestion
      - kind: ConfigMap
        name: cluster-settings-user
      - kind: Secret
        name: cluster-secrets-user
```

File: bootstrap/templates/kubernetes/flux/apps.yaml.j2 [hunk 8]
```diff
@@ -38,6 +42,10 @@ spec:
                 name: cluster-settings
               - kind: Secret
                 name: cluster-secrets
+              - kind: Secret
+                name: cluster-secrets-user
+              - kind: ConfigMap
+                name: cluster-settings-user
```
Comment (onedr0p) [inline bootstrap/templates/kubernetes/flux/apps.yaml.j2:48]:
```suggestion
              - kind: ConfigMap
                name: cluster-settings-user
              - kind: Secret
                name: cluster-secrets-user
```

File: .vscode/settings.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (mrueg) [reply .vscode/settings.json:19]: Maybe add  the .vscode directory to .gitignore? Or is there anything in that users would profit from?

[the remark this answers is not in the document]
Comment (aumer-amr) [reply .vscode/settings.json:19]: I don't really think there are benefits from it is it?

[the remark this answers is not in the document]
Comment (onedr0p) [reply .vscode/settings.json:19]: It is kind of useful e.g. `"sops.defaults.ageKeyFile": "age.key"` is required now for the sops age vscode extension

File: bootstrap/tasks/kubernetes/main.yaml [hunk 9]
```diff
@@ -8,7 +8,43 @@
   with_community.general.filetree: ["../templates/kubernetes/"]
 
 - name: Template Kubernetes unencrypted files
-  when: item.state == 'file' and 'sops' not in item.path and '.DS_Store' not in item.path
+  when:
+    - item.state == 'file'
+    - 'sops' not in item.path
+    - item.path not is match('cluster-(settings|secrets)-user\.yaml\.j2')
```
Comment (onedr0p) [inline bootstrap/tasks/kubernetes/main.yaml:14]:
```suggestion
    - item.path not is regex_match('cluster-(settings|secrets)-user\.yaml\.j2')
```

**Review by onedr0p — APPROVED** [review]

**Comment by aumer-amr** [comment]
🚀

**Pull request merged**